apps/backend/src/app/modules/maps/map-credits.service.ts

```diff
@@ -100,72 +55,99 @@ export class MapCreditsService {
     return DtoFactory(MapCreditDto, dbResponse);
   }
 
-  async updateCredit(
-    mapCreditID: number,
-    creditUpdate: UpdateMapCreditDto,
-    userID: number
-  ): Promise<void> {
-    if (!creditUpdate.userID && !creditUpdate.type)
-      throw new BadRequestException('No update data provided');
-
-    const credit = await this.db.mapCredit.findUnique({
-      where: { id: mapCreditID },
-      include: { user: true, mmap: true }
-    });
-
-    if (!credit || !credit.mmap || !credit.user)
-      throw new NotFoundException('Invalid map credit');
-
-    await this.checkCreditChangePermissions(credit.mmap, userID);
-
-    if (
-      creditUpdate.userID &&
-      !(await this.db.user.exists({
-        where: { id: creditUpdate.userID }
-      }))
-    )
-      throw new BadRequestException('Credited user does not exist');
-
-    // Check for different credits with same map, user and type, throw if exists
-    if (
-      await this.db.mapCredit.exists({
-        where: {
-          NOT: { id: credit.id },
-          userID: creditUpdate.userID ?? credit.userID,
-          type: creditUpdate.type ?? credit.type,
-          mapID: credit.mapID
-        }
-      })
-    )
-      throw new ConflictException('Cannot have duplicate map credits');
-
-    const data: Prisma.MapCreditUpdateInput = {};
-    if (creditUpdate.userID)
-      data.user = { connect: { id: creditUpdate.userID } };
-    if (creditUpdate.type) data.type = creditUpdate.type;
-
-    const newCredit = await this.db.mapCredit.update({
-      where: { id: mapCreditID },
-      data
-    });
+  async updateCredits(
+    mapID: number,
+    body: CreateMapCreditDto[],
+    loggedInUserID: number
+  ): Promise<MapCreditDto[]> {
+    if (body.length === 0) {
+      throw new BadRequestException('Empty body');
+    }
 
-    await this.updateMapCreditActivities(newCredit, credit);
-  }
+    if (!body.some((credit) => credit.type === MapCreditType.AUTHOR))
+      throw new BadRequestException('Credits do not contain an AUTHOR');
 
-  async deleteCredit(mapCreditID: number, userID: number): Promise<void> {
-    const credit = await this.db.mapCredit.findUnique({
-      where: { id: mapCreditID },
-      include: { user: true, mmap: true }
+    const map = await this.db.mMap.findUnique({
+      where: { id: mapID },
+      include: { credits: true }
     });
+    if (!map) {
+      throw new NotFoundException('Map not found');
+    }
 
-    if (!credit || !credit.mmap || !credit.user)
-      throw new NotFoundException('Invalid map credit');
+    await this.checkCreditChangePermissions(map, loggedInUserID);
+
+    // TODO: Wrap in transaction!
```

Comment:
Should this TODO be tackled in this PR? Or should there be an issue?

Reply:
In next PR

Reply:
Required some DB refactoring, couldn't do in this, is in next